pkg/prov/http/svc.go

```diff
@@ -0,0 +1,59 @@
+package http
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ksysoev/deriv-api-bff/pkg/core"
+	"github.com/ksysoev/deriv-api-bff/pkg/core/request"
+	"github.com/ksysoev/wasabi"
+	"github.com/ksysoev/wasabi/backend"
+	"github.com/ksysoev/wasabi/channel"
+)
+
+type Service struct {
+	handler wasabi.RequestHandler
+}
+
+// NewService initializes and returns a new instance of Service.
+// It sets up the handler with a new backend using the requestFactory.
+// It returns a pointer to the newly created Service instance.
+func NewService() *Service {
+	s := &Service{}
+
+	s.handler = backend.NewBackend(s.requestFactory)
+
+	return s
+}
+
+func (s *Service) Handle(conn *core.Conn, req *request.HTTPReq) error {
+	connWrap := channel.NewConnectionWrapper(conn, channel.WithSendWrapper(
+		func(_ wasabi.Connection, _ wasabi.MessageType, msg []byte) error {
+			if ok := conn.DoneRequest(req.ID(), msg); !ok {
+				return fmt.Errorf("request ID %d not found is cancelled", req.ID())
+			}
+
+			return nil
+		},
+	))
+
+	return s.handler.Handle(connWrap, req)
+}
+
+// requestFactory creates an HTTP request from a wasabi.Request.
+// It takes a parameter r of type wasabi.Request.
+// It returns a pointer to an http.Request and an error.
+// It returns an error if the request type is invalid or if the HTTP request creation fails.
+func (s *Service) requestFactory(r wasabi.Request) (*http.Request, error) {
+	req, ok := r.(*request.HTTPReq)
```

Comment:
can we do explicit casting? or add a transformer? in future these two types can diverge a lot

Reply:
As you can see in Handle method, we accept only `request.HTTPReq`, here we have `wasabi.Request` because I have to comply with wasabi interface to be able to use http backend from there.  it should be impossible to get here other type but `request.HTTPReq`